src/Document/Factory/DoctrineDocumentFactory.php

```diff
@@ -78,7 +79,11 @@ public function updateDocument(DocumentInterface $document, array $documentData)
             }
           }
         }
-        $data = $this->documentManager->getHydratorFactory()->getHydratorFor($documentName)->hydrate($document, $documentData);
+        try {
+          $data = $this->documentManager->getHydratorFactory()->getHydratorFor($documentName)->hydrate($document, $documentData);
+        } catch (MappingException $me) {
```

Comment:
Create 
src/Exception/DocumentFactoryException extends AbstractCoreException {
private $documentFactory;
}
src/Exception/DocumentMappingException extends DoctrineFactoryException }
private $document;
}

throw DoctrineMappingException ($this, $document, $em)